packages/context/src/binding.ts

```diff
@@ -370,7 +325,7 @@ export class Binding<T = BoundValue> {
     }
     this.type = BindingType.PROVIDER;
     this._getValue = (ctx, session) => {
-      const providerOrPromise = instantiateClass<Provider<T>>(
+      const providerOrPromise = instantiateClass<Provider<BoundValue>>(
```

Comment:
Why `BoundValue` instead of `T`?

Reply:
This looks like an unintended change, I'll revert it.